lib/internal/Magento/Framework/Mail/Template/TransportBuilderByStore.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Framework\Mail\Template;
+
+use Magento\Framework\Mail\MessageInterface;
+
+class TransportBuilderByStore
+{
+    /**
+     * Message.
+     *
+     * @var \Magento\Framework\Mail\Message
+     */
+    protected $message;
+
+    /**
+     * Sender resolver.
+     *
+     * @var \Magento\Framework\Mail\Template\SenderResolverInterface
+     */
+    private $senderResolver;
+
+    /**
+     * @param MessageInterface $message
+     * @param SenderResolverInterface $senderResolver
+     */
+    public function __construct(
+        MessageInterface $message,
+        SenderResolverInterface $senderResolver
+    ) {
+        $this->message = $message;
+        $this->senderResolver = $senderResolver;
+    }
+
+    /**
+     * Set mail from address by store.
+     *
+     * @param string|array $from
+     * @param string|int $store
+     *
+     * @return $this
+     */
+    public function setFromByStore($from, $store)
```

Comment:
Is there any reason to introduce another class and method instead of just adding an optional parameter `scopeId` to `\Magento\Framework\Mail\Template\TransportBuilder::setFrom`?

Reply:
@ishakhsuvarov, is it allowed? According to the backward compatibility(http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/index.html) seems it is prohibited.
May I add optional parameter scopeId?

Reply:
Correct. I overlooked that for this case